src/appengine/handlers/cron/grouper.py

```diff
@@ -86,6 +86,48 @@ def _get_new_group_id():
   return new_group.key.id()
 
 
+def is_same_variant(variant1, variant2):
+  """Checks for the testcase variants equality."""
+  return (variant1.crash_type == variant2.crash_type and
+          variant1.crash_state == variant2.crash_state and
+          variant1.security_flag == variant2.security_flag)
+
+
+def _group_testcases_based_on_variants(testcase_map):
+  """Group testcases that are associated based on variant analysis."""
+  for testcase_1_id, testcase_1 in testcase_map.items():
+    for testcase_2_id, testcase_2 in testcase_map.items():
+      # Rule: Don't group the same testcase and use different combinations for
+      # comparisons.
+      if testcase_1_id <= testcase_2_id:
+        continue
+
+      # Rule: If both testcase have the same group id, then no work to do.
+      if testcase_1.group_id == testcase_2.group_id and testcase_1.group_id:
+        continue
+
+      # Rule: Check both testcase are under the same project.
+      if testcase_1.project_name != testcase_2.project_name:
```

Comment:
Once you add the `job_type` property, also check if they're equal here. If they are, we can bail because variant analysis is irrelevant.

Reply:
Done.